src/plugins/event-plugins/__tests__/ResourcePlugin.test.ts

```diff
@@ -250,4 +247,24 @@ describe('ResourcePlugin tests', () => {
             })
         );
     });
+
+    test('resource plugin ignores resources based on a custom condition', async () => {
```

Comment:
nit: write test description in format ```when <action> then <expected behavior>```  (see above unit tests for reference).

Reply:
Refactored